quickwit-janitor/src/lib.rs

```diff
@@ -34,17 +36,33 @@ pub use janitor_service::JanitorService;
 pub use self::garbage_collection::{
     delete_splits_with_files, run_garbage_collect, FileEntry, SplitDeletionError,
 };
-use crate::actors::GarbageCollector;
+use crate::actors::{DeleteTaskService, GarbageCollector};
 
 pub async fn start_janitor_service(
     universe: &Universe,
-    _config: &QuickwitConfig, // kept it for retention policy
+    config: &QuickwitConfig, // kept it for retention policy
     metastore: Arc<dyn Metastore>,
+    cluster: Arc<Cluster>,
     storage_uri_resolver: StorageUriResolver,
 ) -> anyhow::Result<JanitorService> {
     info!("Starting janitor service.");
-    let garbage_collector = GarbageCollector::new(metastore, storage_uri_resolver);
+    let garbage_collector = GarbageCollector::new(metastore.clone(), storage_uri_resolver.clone());
     let (_, garbage_collector_handle) = universe.spawn_actor(garbage_collector).spawn();
+    let search_client_pool = SearchClientPool::create_and_keep_updated(
```

Comment:
We need to use the same pool as search etc.
Can you open an issue to make that search pool into some kind of singleton?

(a service perhaps)

Reply:
Actually this is very simple to do, I have just fixed it in the PR.